Add first unit test to config flow for Plum Lightpad

prystupa opened the pull request:
<!--
  You are amazing! Thanks for contributing to our project!
  Please, DO NOT DELETE ANY TEXT from this template! (unless instructed).
-->
## Proposed change
<!-- 
  Describe the big picture of your changes here to communicate to the
  maintainers why we should accept this pull request. If it fixes a bug
  or resolves a feature request, be sure to link to that issue in the
  additional information section.
-->
No changes - only adding first simple unit test to config flow

## Type of change
<!--
  What type of change does your PR introduce to Home Assistant?
  NOTE: Please, check only 1! box! 
  If your PR requires multiple boxes to be checked, you'll most likely need to
  split it into multiple PRs. This makes things easier and faster to code review.
-->

- [ ] Dependency upgrade
- [ ] Bugfix (non-breaking change which fixes an issue)
- [ ] New integration (thank you!)
- [ ] New feature (which adds functionality to an existing integration)
- [ ] Breaking change (fix/feature causing existing functionality to break)
- [x] Code quality improvements to existing code or addition of tests

## Additional information
<!--
  Details are important, and help maintainers processing your PR.
  Please be sure to fill out additional details, if applicable.
-->

- This PR fixes or closes issue: fixes #
- This PR is related to issue: 
- Link to documentation pull request: 

## Checklist
<!--
  Put an `x` in the boxes that apply. You can also fill these out after
  creating the PR. If you're unsure about any of them, don't hesitate to ask.
  We're here to help! This is simply a reminder of what we are going to look
  for before merging your code.
-->

- [x] The code change is tested and works locally.
- [x] Local tests pass. **Your PR cannot be merged unless tests pass**
- [x] There is no commented out code in this PR.
- [x] I have followed the [development checklist][dev-checklist]
- [x] The code has been formatted using Black (`black --fast homeassistant tests`)
- [x] Tests have been added to verify that the new code works.

If user exposed functionality or configuration variables are added/changed:

- [ ] Documentation added/updated for [www.home-assistant.io][docs-repository]

If the code communicates with devices, web services, or third-party tools:

- [ ] The [manifest file][manifest-docs] has all fields filled out correctly.  
      Updated and included derived files by running: `python3 -m script.hassfest`.
- [ ] New or updated dependencies have been added to `requirements_all.txt`.  
      Updated by running `python3 -m script.gen_requirements_all`.
- [ ] Untested files have been added to `.coveragerc`.

The integration reached or maintains the following [Integration Quality Scale][quality-scale]:
<!--
  The Integration Quality Scale scores an integration on the code quality
  and user experience. Each level of the quality scale consists of a list
  of requirements. We highly recommend getting your integration scored!
-->

- [ ] No score or internal
- [ ] 🥈 Silver
- [ ] 🥇 Gold
- [ ] 🏆 Platinum

<!--
  Thank you for contributing <3

  Below, some useful links you could explore:
-->
[dev-checklist]: https://developers.home-assistant.io/docs/en/development_checklist.html
[manifest-docs]: https://developers.home-assistant.io/docs/en/creating_integration_manifest.html
[quality-scale]: https://developers.home-assistant.io/docs/en/next/integration_quality_scale_index.html
[docs-repository]: https://github.com/home-assistant/home-assistant.io

prystupa committed:
add first unit test to config flow for Plum Lightpad

prystupa committed:
add first unit test to config flow for Plum Lightpad (add changed requirements_test_all.txt)

MartinHjelmare commented:
What's the coverage status now for the config flow module? Can we include that module in our coverage calculation? It needs 100% coverage for that.

If you update the `.coveragerc` file and explicitly exclude all the other modules except the `config_flow.py` module, you can then check coverage like this:
```
pytest --cov-report term-missing --cov=homeassistant.components.plum_lightpad tests/components/plum_lightpad
```

https://github.com/home-assistant/core/blob/15165a3c938250ac325e2a9ca7c84dd6b02cb7b0/.coveragerc#L624

prystupa committed:
add first unit test to config flow for Plum Lightpad

prystupa committed:
add first unit test to config flow for Plum Lightpad (bring coverage to 100%)

prystupa commented:
> What's the coverage status now for the config flow module? Can we include that module in our coverage calculation? It needs 100% coverage for that.
> 
> If you update the `.coveragerc` file and explicitly exclude all the other modules except the `config_flow.py` module, you can then check coverage like this:
> 
> ```
> pytest --cov-report term-missing --cov=homeassistant.components.plum_lightpad tests/components/plum_lightpad
> ```
> 
> https://github.com/home-assistant/core/blob/15165a3c938250ac325e2a9ca7c84dd6b02cb7b0/.coveragerc#L624

It was not 100%, but I added missing tests to this PR and it is now reported as 100% for just config_flow.py

MartinHjelmare commented on tests/components/plum_lightpad/test_config_flow.py:
```diff
@@ -0,0 +1,82 @@
+    )
+
+    with patch(
+        "homeassistant.components.test_lightpad.config_flow.PlaceholderHub.authenticate",
```
Please update the patch target path.

The target path should be `"homeassistant.components.plum_lightpad.utils.Plum.loadCloudData"`.

prystupa replied:
updated

MartinHjelmare commented on tests/components/plum_lightpad/test_config_flow.py:
```diff
@@ -0,0 +1,82 @@
+    assert result["type"] == "form"
+    assert result["errors"] == {}
+
+    with patch("plumlightpad.Plum.loadCloudData"), patch(
```
Please update the target path here too.

prystupa replied:
updated

MartinHjelmare commented on tests/components/plum_lightpad/test_config_flow.py:
```diff
@@ -0,0 +1,82 @@
+    """Test configuring the flow using configuration.yaml."""
+    await setup.async_setup_component(hass, "persistent_notification", {})
+
+    with patch("plumlightpad.Plum.loadCloudData"), patch(
```
And here.

prystupa replied:
updated

MartinHjelmare reviewed (changes requested):
We should also test aborting on same username. Add a mock config entry to home assistant and then start the flow with the same username.

https://github.com/home-assistant/core/blob/ca265966e79b1196dca38e628992e1c7b03d3783/tests/common.py#L638

Please update `.coveragerc` as mentioned earlier too.

prystupa committed:
add first unit test to config flow for Plum Lightpad

prystupa committed:
add first unit test to config flow for Plum Lightpad (updated patch path as suggested)

prystupa committed:
add first unit test to config flow for Plum Lightpad (add unit test for abort)

prystupa commented:
> We should also test aborting on same username. Add a mock config entry to home assistant and then start the flow with the same username.
> 
> https://github.com/home-assistant/core/blob/ca265966e79b1196dca38e628992e1c7b03d3783/tests/common.py#L638
> 
> Please update `.coveragerc` as mentioned earlier too.

added test for abort, added config for coverage

MartinHjelmare reviewed (approved):
Great!
